.github/workflows/ci.yaml

```diff
@@ -4,15 +4,33 @@ on:
   push:
     branches:
     - 'main'
-env:
-  go-version: "1.20"
 jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - run: make docker-build
   test:
-    name: Small tests
+    name: Unit tests
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-go@v4
+      with:
+        go-version-file: go.mod
+    - run: make setup
+    - run: make test
+  e2e:
+    name: End-to-End Tests
     runs-on: ubuntu-22.04
     steps:
     - uses: actions/checkout@v3
-    - uses: actions/setup-go@v3
+    - uses: actions/setup-go@v4
       with:
-        go-version: ${{ env.go-version }}
+        go-version-file: go.mod
+    - run: make setup
+      working-directory: e2e
+    - run: make start 
+      working-directory: e2e
     - run: make test
+      working-directory: e2e
```

Comment:
Please add a final new line.
If you are using VSCode, you can set `"files.insertFinalNewline": true`.